src/validation.cpp

```diff
@@ -4070,6 +4070,11 @@ bool ChainstateManager::ProcessNewBlock(const std::shared_ptr<const CBlock>& blo
     if (!ActiveChainstate().ActivateBestChain(state, block)) {
         return error("%s: ActivateBestChain failed (%s)", __func__, state.ToString());
     }
+    BlockValidationState background_sync_state;
+    if (WITH_LOCK(::cs_main, return BackgroundSyncInProgress()) &&
```

Comment:
>seems potentially racy

How so?

Reply:
> > seems potentially racy
> 
> How so?

I don't think there is a real race condition here because if stale a BackgroundSyncInProgress value were used, it wouldn't cause any harm. By "potentially racy" I just meant that any time I see a mutex being unlocked and immediately relocked I have to turn on the part of my brain that thinks about race conditions, so it's nicer to use one lock than worry about what state can change in the gap between locks.